src/gluonts/model/deepar/_estimator.py

```diff
@@ -222,6 +237,18 @@ def derive_auto_fields(cls, train_iter):
             "cardinality": [len(cats) for cats in stats.feat_static_cat],
         }
 
+    def train(
+        self,
+        training_data: Dataset,
+        validation_data: Optional[Dataset] = None,
+        num_workers: Optional[int] = None,
+        num_prefetch: Optional[int] = None,
+        **kwargs,
+    ) -> Predictor:
+        self.input_repr.initialize_from_dataset(training_data)
+        self.output_repr.initialize_from_dataset(training_data)
+        return super().train(training_data)
```

Comment:
I think other arguments should be passed on to `super().train`, right? You can probably capture all but `training_data` in **kwargs

Reply:
Ah yeah, right. Fixed!